### PR TITLE
Library rewrite

### PR DESCRIPTION
**Pull request by Iguins**
GitHub Issue: #

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

**Comment by benventive** [comment]
Don't forget to change the pool from `Container` to `Windows 1809`

File: Src/BiometryService/BiometryHelper.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BiometryService
+{
+	internal static class BiometryHelper
```
[the remark this answers is not in the document]
Comment (Iguins) [reply Src/BiometryService/BiometryHelper.cs:10]: Sure!

File: src/BiometryService/BiometryType.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		/// </summary>
+		Fingerprint,
+		/// <summary>
+		///		Face identifier.
+		/// </summary>
+		Face,
+		/// <summary>
+		///		Managed by the OS
+		/// </summary>
+		FaceOrFingerprint
```
Comment (MatFillion) [inline src/BiometryService/BiometryType.cs:23]: will this bite us in the butt later? (if for example iris scan comes), we could use enum flags instead?

File: src/BiometryService/IBiometryService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		Task<BiometryResult> ValidateIdentity(CancellationToken ct);
+
+		/// <summary>
+		/// Encrypt the string value to an array of byte data
+		/// </summary>
+		/// <param name="ct">The <see cref="CancellationToken" /> to use.</param>
+		/// <param name="keyName">The name of the Key.</param>
+		/// <param name="value">The value to be encrypt.</param>
+		/// <returns>An array of byte.</returns>
+		Task<byte[]> Encrypt(CancellationToken ct, string keyName, string value);
```
Comment (MatFillion) [inline src/BiometryService/IBiometryService.cs:39]: According  to #6 this should be just a Task, no need for the byte array.

File: src/BiometryService/IBiometryService.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+		/// <summary>
+		/// Decodes the array of byte data to a string value
+		/// </summary>
+		/// <param name="ct">The <see cref="CancellationToken" /> to use.</param>
+		/// <param name="keyName">The name of the Key.</param>
+		/// <param name="data">The data to be decrypt.</param>
+		/// <returns>A string</returns>
+		Task<string> Decrypt(CancellationToken ct, string keyName, byte[] data);
```
Comment (MatFillion) [inline src/BiometryService/IBiometryService.cs:48]: We don't need byte[] data do we?

File: Src/BiometryService/BiometryHelper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Iguins) [reply Src/BiometryService/BiometryHelper.cs]: Removed as we don't need it anymore

File: Src/BiometryService/BiometryService.Android.cs [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="BiometryService" /> class.
+        /// </summary>
+        /// <param name="fragmentActivity"></param>
+        /// <param name="applicationContext"></param>
+        /// <param name="dispatcher"></param>
+        /// <param name="promptInfoBuilder"></param>
+        /// <param name="authenticators"></param>
+        public BiometryService(
+            FragmentActivity fragmentActivity,
```
[the remark this answers is not in the document]
Comment (Iguins) [reply Src/BiometryService/BiometryService.Android.cs:54]: checking

File: src/BiometryService/IBiometryService.cs (hunk 3 above)
Comment (Iguins) [reply src/BiometryService/IBiometryService.cs]: Fixed!

**Pull request merged**